rules/lexer.l

```diff
@@ -55,6 +55,11 @@ L                       [a-zA-Z_]
 M                       [a-zA-Z_:]
 U                       [smhdwy]
 
+FLOAT                   [-+]?{D}*\.?{D}+{EXPONENT}?|[+-]?[iI][nN][fF]|[nN][aA][nN]
```

Comment:
A number like `42.` would be illegal with this expression, but it's commonly used to mark a number as float. We should probably be able to parse it, too.